src/classes/Fetcher.py

```diff
@@ -33,48 +37,60 @@ def fetchStockCodes(executeOption):
             listStockCodes = list(nse.get_stock_codes(cached=False))[1:]
             if len(listStockCodes) > 10:
-                print(colorText.GREEN + ("=> Done! Fetched %d stock codes." % len(listStockCodes)) + colorText.END)
+                print(colorText.GREEN + ("=> Done! Fetched %d stock codes." %
+                                         len(listStockCodes)) + colorText.END)
                 if ConfigManager.shuffleEnabled:
                     random.shuffle(listStockCodes)
-                    print(colorText.BLUE + "[+] Stock shuffling is active." + colorText.END)
+                    print(colorText.BLUE +
+                          "[+] Stock shuffling is active." + colorText.END)
                 else:
-                    print(colorText.FAIL + "[+] Stock shuffling is inactive." + colorText.END)
+                    print(colorText.FAIL +
+                          "[+] Stock shuffling is inactive." + colorText.END)
                 if ConfigManager.stageTwo:
-                    print(colorText.BLUE + "[+] Screening only for the stocks in Stage-2! Edit User Config to change this." + colorText.END)
+                    print(
+                        colorText.BLUE + "[+] Screening only for the stocks in Stage-2! Edit User Config to change this." + colorText.END)
                 else:
-                    print(colorText.FAIL + "[+] Screening only for the stocks in all Stages! Edit User Config to change this." + colorText.END)
-                
+                    print(
+                        colorText.FAIL + "[+] Screening only for the stocks in all Stages! Edit User Config to change this." + colorText.END)
+
             else:
-                input(colorText.FAIL + "=> Error getting stock codes from NSE! Press any key to exit!" + colorText.END)
+                input(
+                    colorText.FAIL + "=> Error getting stock codes from NSE! Press any key to exit!" + colorText.END)
                 sys.exit("Exiting script..")
 
     # Fetch stock price data from Yahoo finance
     def fetchStockData(stockCode, period, duration, proxyServer, screenResults):
         global screenCounter
-        with SuppressOutput(suppress_stdout = True, suppress_stderr = True):
+        with SuppressOutput(suppress_stdout=True, suppress_stderr=True):
             data = yf.download(
-                tickers = stockCode+".NS",
-                period = period,
-                duration = duration,
-                proxy = proxyServer,
+                tickers=stockCode+".NS",
+                period=period,
+                duration=duration,
+                proxy=proxyServer,
                 progress=False
             )
         sys.stdout.write("\r\033[K")
         try:
-            print(colorText.BOLD + colorText.GREEN + ("[%d%%] Screened %d, Found %d. Fetching data & Analyzing %s..." % (int(screenCounter/len(listStockCodes)*100), screenCounter, len(screenResults), stockCode)) + colorText.END, end='')
```

Comment:
Is it possible to use `multiprocessing.Value` here to share the state of `screenCounter` and share/increment them among the processes?

Reply:
Yes, It's Possible using ```multiprocessing.Value```.  using ```lock``` on the variable it gets incremented without any race condition.
Thank you for the advice!!!.